exetera/processing/date_time_helpers.py

```diff
@@ -0,0 +1,172 @@
+from typing import Sequence, Tuple, Union
+from datetime import datetime, timedelta
+
+import numpy as np
+from numpy.typing import ArrayLike
+
+from exetera.core.utils import SECONDS_PER_DAY
+
+
+
+def get_periods(start_date: datetime,
+                end_date: datetime,
+                period: str,
+                delta: int = 1
+                ) -> Sequence[datetime]:
+    """
+    Generate a set of periods into which timestamped data can be grouped.
+    Delta controls whether the sequence of periods is generated from an start point
+    or an end point. When delta is positive, the sequence is generated forwards in time.
+    When delta is negative, the sequence is generate backwards in time.
+    :param start_date: a ``datetime.datetime`` object for the starting period
+    :param end_date: a ``datetime.datetime`` object for tne ending period, exclusive
+    :param period: a string representing the unit in which the delta is calculated
+    ('day', 'days', 'week', 'weeks')
+    :param delta: an integer representing the delta.
+    :return:
+    """
+
+    period_map = {
```

Comment:
```python
if period in ("day","days"):
    tdelta=timedelta(days=delta)
elif period in ("week","weeks"):
    tdelta=timedelta(weeks=delta)
else:
    raise ValueError(f"Unknown period specifier '{period}'")
```

and remove lines 37-8, 50.

Reply:
I envisage this being expanded to handle things like calendar months and so forth. I'd prefer to keep it as a dictionary as I think it is a cleaner way to represent it